compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -976,11 +992,11 @@ impl<'tcx> Term<'tcx> {
         }
     }
 
-    /// This function returns `None` for `AliasKind::Opaque`.
+    /// This function returns the inner `AliasTy` if this term is a projection.
     ///
     /// FIXME: rename `AliasTy` to `AliasTerm` and make sure we correctly
     /// deal with constants.
-    pub fn to_alias_term_no_opaque(&self, tcx: TyCtxt<'tcx>) -> Option<AliasTy<'tcx>> {
+    pub fn to_projection_term(&self, tcx: TyCtxt<'tcx>) -> Option<AliasTy<'tcx>> {
```

Comment:
I'm not a huge fan of this rename, when we start representing free type aliases in `Ty` I would expect this function to return `Some` for them. It's also not really accurate for `ConstKind::Unevaluated` which represents anon consts which are sort of like free consts (although nothing about normalization or equality of `ConstKind::Unevaluated` is really implemented right for new solver yet so shrug)

Reply:
I just think the old name is unnecessarily obtuse though, and also kinda long -- I'd rather have the behavior not be defined by what it is "not" if possible. If you can think of a better one, though, happy to consider naming it something else.